packages/deployments/contracts/contracts/core/connext/helpers/RelayerProxyHub.sol

```diff
@@ -71,6 +82,42 @@ contract RelayerProxyHub is RelayerProxy {
     bytes[] memory _encodedData,
     uint256 _relayerFee
   ) external onlyRelayer nonReentrant {
+    uint256 sum = _propagate(_connectors, _messageFees, _encodedData);
+    emit FundsDeducted(sum, address(this).balance);
+    transferRelayerFee(_relayerFee);
+  }
+
+  // Returns a boolean that indicates if a job is workable or not.
+  function propagateWorkable() public view returns (bool _isWorkable) {
+    return block.timestamp > (lastPropagateAt + propagateCooldown);
+  }
+
+  function propagateKeep3r(
+    address[] calldata _connectors,
+    uint256[] calldata _messageFees,
+    bytes[] memory _encodedData
+  ) external validateAndPayWithCredits(msg.sender) nonReentrant {
+    require(propagateWorkable(), "Job is not workable");
+    _propagate(_connectors, _messageFees, _encodedData);
```

Comment:
Maybe good idea to imitate the logic above?

```suggestion
    uint256 sum = _propagate(_connectors, _messageFees, _encodedData);
    lastPropagateAt = block.timestamp;
    emit FundsDeducted(sum, address(this).balance)
```

Reply:
There will be no sum here though because there's no synchronous fee being paid from the contract itself.

Reply:
ah yes, my bad--comment can be resolved